docs/content/Dialog2.mdx

```diff
@@ -0,0 +1,112 @@
+3. If there are buttons in the dialog footer, focus can be moved between them with left and right arrow keys or tab/shift+tab.
+
+### Custom rendering
+
+**Note: using custom rendering allows break out of the defined design, UX, and accessibility patterns of the dialog. Use only as a last resort.**
+
+By default, the Dialog component implements the design and interactions defined by the Primer design system. If necessary, you can provide custom renderers for the header, body, or footer using the `renderHeader`, `renderBody`, and `renderFooter` props, respectively. The JSX produced by these render props will render full-bleed into their respective areas of the dialog. If you are using the custom renderers, you should use the provided sub-components `Dialog.Header`, `Dialog.Title`, `Dialog.Subtitle`, `Dialog.CloseButton`, `Dialog.Body`, `Dialog.Footer`, and `Dialog.Buttons` to the extent possible.
+
+### Example
+
+```jsx live
+<State default={false}>
+  {([isOpen, setIsOpen]) => {
+    const returnFocusRef = React.useRef(null)
+    const openDialog = React.useCallback(() => setIsOpen(true), [setIsOpen])
+    const closeDialog = React.useCallback(() => setIsOpen(false), [setIsOpen])
+    return (
+      <>
+        <Button ref={returnFocusRef} onClick={openDialog}>
+          Open
+        </Button>
+        {isOpen && (
+          <Dialog2
+            title="Dialog example"
+            subtitle={
+              <>
+                This is a <b>description</b> of the dialog.
+              </>
+            }
+            footerButtons={[{text: 'Ok', onClick: closeDialog}]}
+            onClose={closeDialog}
+          >
+            <Text fontFamily="sans-serif">Some content</Text>
+          </Dialog2>
+        )}
+      </>
+    )
+  }}
+</State>
+```
+
+## System props
+
+`Dialog`, `Dialog.Header`, `Dialog.Title`, `Dialog.Subtitle`, `Dialog.Body`, and `Dialog.Footer` components get the `COMMON`, `LAYOUT`, and `FLEX` categories of system props. Read our [System Props](/system-props) doc page for a full list of available props.
+
+## Component props
+
+### DialogProps
+
+| Prop name     | Type                                        | Default            | Description                                                                                                   |
+| :------------ | :------------------------------------------ | :----------------- | :------------------------------------------------------------------------------------------------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| title         | `string`                                    | `"Dialog"`         | Required. Sets the title of the dialog, which by default is also used as the `aria-labelledby` attribute.     |
```

Comment:
Is this `Required` if there is a default?

Reply:
I think what I did on the others is not say "Required" or "Optional", so I'll make that change here.